spark/src/main/scala/org/apache/comet/serde/QueryPlanSerde.scala

```diff
@@ -2626,6 +2645,11 @@ object QueryPlanSerde extends Logging with ShimQueryPlanSerde with CometExprShim
           return None
         }
 
+        if ((join.leftKeys ++ join.rightKeys).exists(_.dataType.isInstanceOf[StructType])) {
+          withInfo(join, "Unsupported struct data type in join keys")
+          return None
+        }
```

Comment:
This is an existing issue, but because of us supporting more expressions this caused failures in one of the spark test cases.

Reply:
Seems like we this should be resolved in future versions of datafusion. With https://github.com/apache/datafusion/pull/11232 taking care of the last issues.

Reply:
Created this issue https://github.com/apache/datafusion-comet/issues/621 so re remember removing the guard in the future.